ast/src/parsed/display.rs

```diff
@@ -864,4 +916,124 @@ mod tests {
         assert_eq!(p.clone().join(s.clone()).to_string(), "::abc::y");
         assert_eq!(SymbolPath::from(p.join(s)).to_string(), "::abc::y");
     }
+
+    mod binary_op_parentheses {
```

Comment:
I think the tests are much better readable if the AST is parsed instead of constructed.
You can take a look at parser/lib/src.rs. Maybe it makes sense to create a new file in the test directory inside the parser crate, though.
Then the tests would look like this:
```
#[test]
fn add_mul_paren() {
        let input = "let t = ((x + y) * z);"
        let printed = format!("{}", parse(None, input).unwrap_err_to_stderr());
        assert_eq!("let t = (x + y) * z;", printed);
}
```

Reply:
I added tests there but kept this one as a reference. We can add further tests in parser/tests/parentheses_test.rs